sdk/core/azure-core/src/http/websockets/websockets_impl.hpp

```diff
@@ -0,0 +1,372 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+#include "azure/core/http/websockets/websockets.hpp"
+#include "azure/core/http/websockets/websockets_transport.hpp"
+#include "azure/core/internal/diagnostics/log.hpp"
+#include "azure/core/internal/http/pipeline.hpp"
+#include <array>
+#include <queue>
+#include <random>
+#include <shared_mutex>
+#include <thread>
+
+// Implementation of WebSocket protocol.
+namespace Azure { namespace Core { namespace Http { namespace WebSockets { namespace _detail {
+
+  // Generator for random bytes. Used in WebSocketImplementation and tests.
+  std::vector<uint8_t> GenerateRandomBytes(size_t vectorSize);
+
+  class WebSocketImplementation {
+    enum class SocketState
```

Comment:
SocketStateType?

Reply:
This enum defines a set of states the socket can be in. It doesn't define a type or kind of socket.